query/executor.go

```diff
@@ -289,6 +292,98 @@ func (e *Executor) WithLogger(log *zap.Logger) {
 	e.TaskManager.Logger = e.Logger
 }
 
+func initQueryLogWriter(log *zap.Logger, e *Executor, path string) (*os.File, error) {
+	logFile, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0644)
+	if err != nil {
+		e.Logger.Error("failed to open log file", zap.Error(err))
+		return nil, err
+	}
+
+	existingCore := log.Core()
+
+	encoderConfig := zap.NewProductionEncoderConfig()
+
+	fileCore := zapcore.NewCore(
+		zapcore.NewJSONEncoder(encoderConfig),
+		zapcore.Lock(logFile),
+		zapcore.InfoLevel,
+	)
+
+	newCore := zapcore.NewTee(existingCore, fileCore)
+	e.Logger = zap.New(newCore)
+	e.TaskManager.Logger = e.Logger
+
+	return logFile, nil
+}
+
+func closeQueryLogWriter(fd uintptr, e *Executor) {
```

Comment:
I wouldn't operate on the `fd` here.  Stick with the higher level libraries, like `os.File.Sync` and `os.File.Close`

Reply:
But I might call our utility function [SyncDir](https://github.com/influxdata/influxdb/blob/694607a22c190ed648de46ef029ef2b74d2cc064/pkg/file/file_unix.go#L11) on the containing directory to ensure the log files are saved in case of bad crashes.